File: leetcode/Easy/0459-repeated-substring-pattern/0459-repeated-substring-pattern.py
class Solution:
    def repeatedSubstringPattern(self, s: str) -> bool:
        n = len(s)
        for i in range(1, n // 2 + 1):
            if n % i == 0:
                sub = s[:i] * (n // i)
                if s == sub:
                    return True
        return False

File: leetcode/Easy/0459-repeated-substring-pattern/0459-repeated-substring-pattern.cpp
class Solution {
public:
    bool repeatedSubstringPattern(string s) {
        int n = s.size();
        for(int i=1;i<n/2+1;i++){
            if (n%i == 0){
                string sub = "";
                for(int j=0;j<n/i;j++)
                    sub += s.substr(0,i);
                
                if(sub == s)
                    return true;
            }
        }
        return false;
    }
};